quadriga/src/main/java/edu/asu/diging/quadriga/web/ExploreCollectionController.java

```diff
@@ -0,0 +1,163 @@
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.model.DefaultMapping;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.model.MappedTripleType;
+import edu.asu.diging.quadriga.core.model.mapped.Concept;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+import edu.asu.diging.quadriga.core.service.MappedTripleService;
+import edu.asu.diging.quadriga.web.model.GraphElements;
+
+@Controller
+public class ExploreCollectionController {
+
+    Logger logger = LoggerFactory.getLogger(getClass());
+
+    private static final String URI_PREFIX = "http";
+    private static final String URI_PREFIX_1 = "https";
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    @Autowired
+    private MappedTripleGroupService mappedTripleGroupService;
+
+    @Autowired
+    private MappedTripleService mappedTripleService;
+    
+    @Autowired
+    private ConceptCacheService conceptCacheService;
+    
+    @Autowired
+    private ConceptService conceptService;
+  
+    @RequestMapping(value = "/auth/collections/{collectionId}/explore")
+    public String exploreTriples(@PathVariable String collectionId, Model model) {
+        Collection collection;
+        try {
+            collection = collectionManager.findCollection(collectionId);
+        } catch (InvalidObjectIdException e) {
+            logger.error("No Collection found for id {}", collectionId);
+            return "error404Page";
+        }
+        model.addAttribute("collectionName", collection.getName());
+        model.addAttribute("collection", collectionId);
+        return "auth/exploreCollection";
+    }
+
+    @GetMapping(value = "/auth/collections/{collectionId}/sub-graph")
+    public ResponseEntity<GraphElements> getGraphForUri(@PathVariable String collectionId,
+            @RequestParam(value = "uri", required = true) String uri,
+            @RequestParam(value = "ignoreList", required = false, defaultValue = "{}") List<String> ignoreList)
+            throws InvalidObjectIdException,CollectionNotFoundException{
+        GraphElements graphElements = new GraphElements();
+        try {
+            ConceptCache conceptCache = conceptCacheService.getConceptByUri(uri);
+            MappedTripleGroup mappedTripleGroup = mappedTripleGroupService.findByCollectionIdAndMappingType(collectionId, MappedTripleType.DEFAULT_MAPPING);
+            List<DefaultMapping> triples= mappedTripleService.getTriplesByUri(mappedTripleGroup.get_id().toString(),
+                    mapConceptUriToDatabaseUri(mappedTripleGroup.get_id().toString(),conceptCache.getEqualTo()), ignoreList);
+            
+            graphElements = GraphUtil.mapToGraph(triples);
```

Comment:
there is too much business logic going on in the controller; most of it should be in a service/manager class.